api/lib/models/checkout-object/checkout-object-data-mapper-impl.ts

```diff
@@ -0,0 +1,193 @@
+    @Inject('BunyanLogger') protected readonly logger: Logger,
+  ) {
+    super(acl);
+    super.addRBAC(
+      [this.READ, this.READ_ALL, this.CREATE, this.UPDATE, this.DELETE],
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    );
+  }
+  public delete(object: CheckoutObject): Promise<IDbResult<void>> {
+    const query = squel.delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, object.id)
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    ).pipe(
+      map(() => ({ result: 'Success', data: undefined })),
+    ).toPromise();
+  }
+
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<CheckoutObject>> {
+    let queryBuilder = squel.select({ 
+      autoQuoteFieldNames: true, 
+      autoQuoteTableNames: true 
+    })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<CheckoutObject>(query.text, query.values, { stream: false, cache: true }))
+      .pipe(
+        map((checkoutObject: CheckoutObject) => ({ result: 'Success', data: checkoutObject })),
+      )
+      .toPromise();
+  }
+
+  public async getAll(opts?: IUowOpts): Promise<IDbResult<Stream<CheckoutObject>>> {
+    let queryBuilder = squel.select({
+      autoQuoteFieldNames: true,
+      autoQuoteTableNames: true,
+    })
+      .from(this.tableName, 'checkoutObject');
+      if(opts && opts.fields) {
+        queryBuilder = queryBuilder.fields(opts.fields);
+      }
+      if(opts && opts.startAt) {
+        queryBuilder = queryBuilder.offset(opts.startAt);
+      }
+      if(opts && opts.count) {
+        queryBuilder = queryBuilder.limit(opts.count);
+      }
+      if (opts && opts.byHackathon) {
+        queryBuilder = queryBuilder.
+            where(
+              'hackathon_id = ?',
+              await (opts.hackathon ?
+                Promise.resolve(opts.hackathon) :
+                this.activeHackathonDataMapper.activeHackathon.pipe(map(hackathon => hackathon.uid)).toPromise()),
+            );
+      }
+      const query = queryBuilder
+        .toString()
+        .concat(';');
+      return from(this.sql.query<CheckoutObject>(query, [], { stream: true, cache: true}))
+        .pipe(
+          map((checkoutObjectStream: Stream<CheckoutObject>) => ({ result: 'Success', data: checkoutObjectStream }))
+        )
+        .toPromise();
+  }
+
+  /**
+   * Returns a count of the number of CheckoutObject objects.
+   * @returns {Promise<Readable>}
+   */
+  public async getCount(opts?: IUowOpts): Promise<IDbResult<number>> {
```

Comment:
This function is not `await`ing anything. Does it have to be async?

Reply:
Nope, will change